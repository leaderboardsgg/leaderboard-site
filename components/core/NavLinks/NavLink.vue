<template>
  <BaseButtonLink
    class="nav-link"
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </BaseButtonLink>
</template>

<style scoped>
  @layer components {
    .nav-link {
      @apply w-full mt-2 lg:mt-0 lg:mx-1 py-1 hover:bg-gray-100;
    }
  }
</style>
