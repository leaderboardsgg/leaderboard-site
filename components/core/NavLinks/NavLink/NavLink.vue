<template>
  <BaseButtonLink class="nav-link" v-bind="$attrs" v-on="$listeners">
    <slot />
  </BaseButtonLink>
</template>

<style scoped>
.nav-link {
  @apply w-full mt-2 lg:mt-0 lg:mx-1 py-1 hover:bg-gray-100;
}
</style>
