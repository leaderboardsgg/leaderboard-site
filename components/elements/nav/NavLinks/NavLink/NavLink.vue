<script setup lang="ts">
import ButtonLink from 'elements/buttons/ButtonLink/ButtonLink.vue'
</script>

<template>
  <ButtonLink class="nav-link" v-bind="$attrs">
    <slot />
  </ButtonLink>
</template>

<style lang="postcss" scoped>
.nav-link {
  @apply w-full mt-2 lg:mt-0 lg:mx-1 py-1 hover:bg-gray-100;
}
</style>
