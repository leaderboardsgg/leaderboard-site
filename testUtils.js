import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
