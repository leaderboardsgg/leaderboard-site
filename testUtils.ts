import { createElementBlock, Component, ComponentOptions } from 'vue'
import { MountingOptions } from '@vue/test-utils'
import userEvent from '@testing-library/user-event'
import { render, RenderOptions } from '@testing-library/vue'

export const stubbedRender = (
  Component: Component | ComponentOptions,
  options?: RenderOptions | MountingOptions<any>,
) => {
  return render(Component, {
    global: {
      stubs: {
        NuxtLink: createElementBlock('a'),
      },
    },
    ...options,
  })
}

export * from '@testing-library/vue'
export * from '@nuxt/test-utils'
export * from '@vue/test-utils'
export { userEvent as fireEvent }
