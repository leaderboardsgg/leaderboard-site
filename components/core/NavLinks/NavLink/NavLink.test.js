import { stubbedRender } from '@/testUtils';
import NavLink from './NavLink';

describe('<NavLink />', () => {
  const defaultAttrs = { class: 'custom-link' };
  const defaultProps = { to: '/games' };

  it('should render without crashing', () => {
    const { unmount } = stubbedRender(NavLink, {
      attrs: defaultAttrs,
      props: defaultProps,
      slots: { default: 'Games' },
    });

    unmount();
  });

  it('renders correctly', () => {
    stubbedRender(NavLink, {
      attrs: defaultAttrs,
      props: defaultProps,
      slots: { default: 'Games' },
    });

    expect(document.querySelector('.custom-link')).toMatchSnapshot();
  });
});
