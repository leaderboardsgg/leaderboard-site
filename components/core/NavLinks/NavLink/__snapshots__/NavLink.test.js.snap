// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> renders correctly 1`] = `
<a
  class="custom-link"
  href="/games"
  to="/games"
>
  Games
</a>
`;
